pkg/stub/action/integration/util.go

```diff
@@ -24,7 +25,81 @@ func LookupContextForIntegration(integration *v1alpha1.Integration) (*v1alpha1.I
 		return &ctx, nil
 	}
 
-	return nil, nil
+	ctxList := v1alpha1.NewIntegrationContextList()
+	if err := sdk.List(integration.Namespace, &ctxList); err != nil {
+		return nil, err
+	}
+
+	var c *v1alpha1.IntegrationContext
+
+	if len(integration.Spec.Dependencies) == 0 {
+		// The integration has no dependencies, try to find the one that
+		// has minimum dependencies requirement.
+
+		ndeps := math.MaxUint16
```

Comment:
The final rule should be:
- Take "X" if has the same set of dependencies
- If no context has the same set, then create a context from the most similar one (advanced) or just create a new one
- Don't use context "Y" if there is even just 1 library in "Y" that is not required by the integration (a.k.a. don't put unrelated garbage in the JVM)

Wdyt?

Reply:
Sounds good and simpler